src/main/java/seedu/address/logic/commands/todolistcommands/EditTaskCommand.java

```diff
@@ -92,10 +94,10 @@ private static Task createEditedTask(Task taskToEdit, EditTaskDescriptor editTas
 
         Task editedTask = new Task(name);
 
-        if (taskToEdit.getTag().isPresent() || editTaskDescriptor.getTag().isPresent()) {
-            Tag updatedTag = editTaskDescriptor.getTag()
-                    .orElse(taskToEdit.getTag().get());
-            editedTask = editedTask.setTag(updatedTag);
+        if (taskToEdit.getTags().isPresent() || editTaskDescriptor.getTags().isPresent()) {
+            Set<Tag> updatedTag = editTaskDescriptor.getTags()
+                    .orElse(taskToEdit.getTags().get());
```

Comment:
I think that there might be a NoSuchElement exception thrown for certain cases. This is also applciable when the same check is performed for Date and Priority. This is because these are optional fields and even if the editTaskDescriptor has Tags present, when the argument of the orElse method is executed, i.e. taskToEdit.getTags().get(), the optional might be empty.

Reply:
But if the tags are present in editTaskDescriptor, then orElse() will not execute though? Sine it will only execute if the tags is empty?

Reply:
I will change from using the `orElse()` method to `orElseGet()` method. This way it will not cause a `NoSuchElementException` since we are using a supplier that will call the `get()` method only if the tags in `taskToEdit` is present (unlike `orElse()` that will call `get()` regardless if the tags is present or not).

Reply:
Okay, I have pushed the commit. I will do some further testing after the storage component is fixed 😃

Reply:
Yes, that looks good to me!